rule/confusing-naming.go

```diff
@@ -136,11 +136,12 @@ func getStructName(r *ast.FieldList) string {
 
 	t := r.List[0].Type
 
-	switch v := t.(type) {
-	case *ast.StarExpr:
-		t = v.X
-	case *ast.IndexExpr:
-		t = v.X
+	if p, _ := t.(*ast.StarExpr); p != nil {
+		t = p.X
+	}
+
+	if p, _ := t.(*ast.IndexExpr); p != nil {
+		t = p.X
```

Comment:
Preserving the original switch would be more idiomatic and let us avoiding type assertions multiple times.

Reply:
I tried to avoid a lot of code nesting. Returned the switch